xbmc/music/MusicDatabase.cpp

```diff
@@ -1194,6 +1228,53 @@ bool CMusicDatabase::UpdateArtist(const CArtist& artist)
   return true;
 }
 
+int CMusicDatabase::AddArtist(const std::string& strArtist, const std::string& strMusicBrainzArtistID, const std::string& strSortName)
+{
+  std::string strSQL, strArtistName, strArtistSort, strNewSort;
+  int idArtist = AddArtist(strArtist, strMusicBrainzArtistID);
+  if (idArtist < 0 || strSortName.empty())
+    return idArtist;
+  
+  /* Artist sort name always taken as the first value provided that is different from name, so only  
+     update when current sort name is blank. If a new sortname the same as name is provided then
+     clear any sortname currently held.
+  */ 
+
+  try
+  {
+    if (NULL == m_pDB.get()) return -1;
+    if (NULL == m_pDS.get()) return -1;
+
+    strSQL = PrepareSQL("SELECT strArtist, strSortName FROM artist WHERE idArtist = %i", idArtist);
+    m_pDS->query(strSQL);
+    if (m_pDS->num_rows() != 1)
+    {
+      m_pDS->close();
+      return -1;
+    }
+    strArtistName = m_pDS->fv("strArtist").get_asString();
```

Comment:
I would move the declarations here as they aren't used outside the try block there's no real reason to have a separate declaration for it. Makes a tiny bit harder to know if they're actually empty or something is overwritten.